microprofile/telemetry-smallrye/extension/src/main/java/org/wildfly/extension/microprofile/telemetry/MicroProfileTelemetrySubsystemAdd.java

```diff
@@ -36,12 +30,6 @@ protected void performBoottime(OperationContext context, ModelNode operation, Mo
 
         super.performBoottime(context, operation, model);
 
-        ServiceInstaller.builder(ServiceDependency.on(WildFlyOpenTelemetryConfig.SERVICE_DESCRIPTOR))
-                .withCaptor(openTelemetryConfig::set)
-                .build()
-                .install(context);
```

Comment:
This installs a "passive" service that captures the WildFlyOpenTelemetryConfig as soon as the service the provides it is started.  While this correctly makes an existing WildFlyOpenTelemetryConfig available to the MicroProfileTelemetryDeploymentProcessor, it is do not _require_ that one is available.  If the WildFlyOpenTelemetryConfig is meant to be a requirement, then this mechanism is insufficient for this use case.  I assume that is the case here?

Reply:
Yes, the WildFlyOpenTelemetryConfig is required. There's also no guarantee that the passive service here is started before the DeploymentUnitPhaseService that runs MicroProfileTelemetryDeploymentProcessor is started.